lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -751,6 +753,47 @@ def handle_module_operation(node, operation)
       end
     end
 
+    sig { params(node: Prism::CallNode).void }
+    def handle_module_function(node)
+      arguments_node = node.arguments
+      return unless arguments_node
+
+      owner_name = @owner_stack.last&.name
+      return unless owner_name
+
+      arguments_node.arguments.each do |argument|
+        method_name = case argument
+        when Prism::StringNode
+          argument.content
+        when Prism::SymbolNode
+          argument.value
+        end
+        next unless method_name
+
+        entries = @index.resolve_method(method_name, owner_name)
+        next unless entries
+
+        entries.each do |entry|
+          entry_owner_name = entry.owner&.name
+          next unless entry_owner_name
+
+          entry.visibility = Entry::Visibility::PRIVATE
+
+          singleton = @index.existing_or_new_singleton_class(entry_owner_name)
+          @index.add(Entry::Method.new(
+            method_name,
+            @file_path,
+            Location.from_prism_location(node.location, @code_units_cache),
+            Location.from_prism_location(T.must(node.message_loc), @code_units_cache),
+            collect_comments(node),
```

Comment:
Typically, developers will document the original method and not the call to `module_function`. For example:

```ruby
module Foo
  # documentation
  def bar; end

  module_function :bar
end
```

By collecting comments on the call node for `module_function`, we are not bringing the documentation associated with the original method.

I think here we want to concatenated the documentation for the existing entry with any other comments that exist on this call node.

```suggestion
            collect_comments(node).concat(entry.comments),
```

Reply:
@vinistock, Upon adding the concat, I changed the test by giving a doc to `foo`. I expected only to receive "This is foo" for the `foo` entry method. However, I also got the "This is foo" comment for the `bar` method entry. Still trying to figure out what's wrong happening!

```ruby
index(<<~RUBY)
  module Test
    # This is foo
    def foo; end
    def bar; end
    module_function :foo, "bar"
  end
RUBY

entries = T.must(@index["foo"])
first_entry, second_entry = *entries
first_entry.comments  # This is foo
second_entry.comments # This is foo

# Not what I expected
entries = T.must(@index["bar"])
first_entry, second_entry = *entries
first_entry.comments  # This is foo
second_entry.comments # This is foo
```

Reply:
There's a chance this is an existing bug. If you add a line break between the `foo` and `bar` declaration, does the issue go away?

Reply:
Please do! You discovered it.